jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -72,6 +74,18 @@ public static void extract(Path source, Path destination) throws IOException {
           }
         }
       }
+
+      // Preserve source modification timestamps of files. If the entry is a symbolic link then set
+      // it's modification time to that of the target file.
+      for (TarArchiveEntry entry : modificationTimeMap.keySet()) {
+        FileTime sourceModificationTime = modificationTimeMap.get(entry);
+        if (entry.isSymbolicLink()) {
+          Path targetPath = destination.resolve(entry.getName()).toRealPath();
+          Path targetRelativePath = destination.relativize(targetPath);
+          sourceModificationTime = modificationTimeMap.get(targetRelativePath);
```

Comment:
`modificationTimeMap.get(targetRelativePath)` doesn't make sense. The key should be of type `TarArchiveEntry` but here it's `Path`. How can this even work?